code/controllers/Processes/shuttles.dm

```diff
@@ -215,3 +219,116 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 				S.dwidth = M.dwidth
 				S.dheight = M.dheight
 		moveShuttle(M.id, "[M.roundstart_move]", 0)
+
+/datum/transit_allocator
+	var/x1
+	var/x2
+	var/y1
+	var/y2
+	var/z
+	var/list/allocated = list()
+
+/datum/transit_allocator/New()
+	for(var/obj/effect/landmark/L in landmarks_list)
```

Comment:
this is incredibly gross
make the transit area coords a config option please, or automatically determine it from where transit turfs are placed